src/google/protobuf/repeated_field.h

```diff
@@ -1255,10 +1255,12 @@ void RepeatedField<Element>::Reserve(int new_size) {
   }
   // Likewise, we need to invoke destructors on the old array. If Element has no
   // destructor, this loop will disappear.
-  e = &old_rep->elements[0];
-  limit = &old_rep->elements[current_size_];
-  for (; e < limit; e++) {
-    e->Element::~Element();
+  if (current_size_ > 0) {
```

Comment:
Maybe simpler just to write:

for (size_t i = 0; i < current_size_; i++) {
  Element* e = old_rep->elements[i];
  e->Element::~Element();
}

Also doesn't some of the code above have the same problem if new_size == 0 (that doesn't make a lot of sense to do, but it shouldn't trigger undefined behavior).